### ISSUE TITLE
Take notes in markdown

### ISSUE THREAD

**Issue by TaylanTatli** [giuspen/cherrytree#218]
Is it possible? I can use * to create bullet lists so maybe it can be possible for others. Actually all I want is using # for headers ``` for code blocks and []() for links.

**Comment by giuspen** [comment]
Hi, currently (0.37.6) links are already supported when surrounded by double square brackets. I plan to add soon more support for markdown typing

**Comment by HaoZeke** [comment]
Any updates on this? (Sorry if it seems like an ETA)

**Comment by Os3r3um** [comment]
Does 0.38.4 version support markdown now?

**Comment by giuspen** [comment]
Nope

**Comment by mani-zaeim** [comment]
First of all I want to appreciate for such great effort to make such great note taking application, I am using it almost everyday for all my technical and none-technical notes, It would be great if you can add makdown support, I just took a look on BoostNote and Typora, one of the cool features beside the Markdown support both added is supporting for some UML Diagram with markdown syntax by using third-party library, boostNote has interesting look and feel well as Syntax highlighter on the code block, it is very user friendly, I really wish we can have it in CherryTree.
Unfortunately I dont have much knowledge on Python; otherwise, I would try to help you on the development.
Again, Thanks for all of your effort.

**Comment by ma-sadeghi** [comment]
Thanks for this great piece of software. Would really appreciate it if you could add markdown support. Thanks :)

**Comment by silverark** [comment]
+1 Markdown support

**Comment by rmfatemi** [comment]
It's awesome. I wish you could add markdown support.

**Comment by DiagonalArg** [comment]
While I'm enthusiastic about the idea, it seems to me markdown has the problem that to do this, Cherrytree will need to have two states: rendered and unrendered (eg, the text).  How do we imagine this?  A hotkey that flips from one to the other?  (That would be slow for large notes.)  Or a split window with rendered below and the editor with unrendered text, above?

It's possible I'm missing something.  In which case, I'm sure someone will tell me.

**Comment by dvdgsng** [comment]
@DiagonalArg That's correct, all of those need to be implemented, that's how all markdown editors I've ever used do it and that's what user expect. Text-Only-View, Preview-Only-View, Text-Preview-SplitView. And of course buttons & hotkeys to switch between.

**Comment by advaithm** [comment]
any news on if this is in development? i would love markdown in cherrytree

**Comment by 0xcrypto** [comment]
#579 and  #55 are much more appropriate features to have than this one.

**Comment by DiagonalArg** [comment]
@0xcrypto - Appropriate?  

I'm not sure I agree with that.  

On the other hand, I do agree that import/export are reasonable first steps.

**Comment by wernerhertzog** [comment]
This is a great note taking program and markdown support would make it simply the best.

**Comment by golubs** [comment]
Some way to write in markdown, automatically convert to HTML, and then export back to markdown again, if needed, would be amazing. I don't know if that makes sense.

**Comment by a56z** [comment]
yes this would be amazing, especially because it would allow to upload larger CherryTree projects to Github Pages for example. 

I would really love to see this implemented

**Comment by Biokore** [comment]
Hello, would be great to get Markdown with maths formulas support as well.

**Comment by DiagonalArg** [comment]
> Hello, would be great to get Markdown with maths formulas support as well.

This was the reason for my request, [here](https://github.com/giuspen/cherrytree/issues/878).  Besides the OP, see also [here](https://github.com/giuspen/cherrytree/issues/878#issuecomment-639536830).

**Comment by wattahay** [comment]
> While I'm enthusiastic about the idea, it seems to me markdown has the problem that to do this, Cherrytree will need to have two states: rendered and unrendered (eg, the text). How do we imagine this? A hotkey that flips from one to the other? (That would be slow for large notes.) Or a split window with rendered below and the editor with unrendered text, above?
> 
> It's possible I'm missing something. In which case, I'm sure someone will tell me.

Zim wiki has a near ideal situation in my experience. You can type in markdown, and it automatically converts. I definitely want at least a keystroke to change back to markdown on at least a per-line or per-heading/paragraph basis.

Although I realize this might be MORE difficult to develop, This could possibly be easier or an interim compromise in development difficulty as a feature request.

**Comment by hackerbabulalala** [comment]
@giuspen  Sir Can we have markdown-style code blocks in cherry-tree. 
just same as we have on Github.

**Comment by hackerbabulalala** [comment]
@giuspen Will cherry tree add features of exporting notes to markdown format???

**Comment by moashxer** [comment]
will there be markdown support in near updates like  for code like below ;
`single code line`
```md
multi
line
```

**Comment by rwperrott** [comment]
Native markdown should be an option, as a complement/support for MD import/export, because it can render _some_ stuff better than rich text, like code sections, and maybe a better target for HTML pasting, which can end up ugly in rich text, so need lots of editing. It would be nice if I could paste/edit MD format directly in a leaf, and be able to switch to/from rendered view.

**Comment by DiagonalArg** [comment]
> Native markdown should be an option, as a complement/support for MD import/export, because it can render some stuff better than rich text, like code sections, and maybe a better target for HTML pasting, which can end up ugly in rich text, so need lots of editing.  It would be nice if I could paste/edit MD format directly in a leaf,

Good point.  This is also [true with LaTeX](https://www.fabriziomusacchio.com/blog/2021-08-10-How_to_use_LaTeX_in_Markdown/), which those of us who do any math, need.

> and be able to switch to/from rendered view.

This is a different issue.  Right now CherryTree offers a WYSIWYG RichText editor with import/export options.  You're asking to add a WYSIWYG Markdown editor (or maybe a hotkey to swap between, or a separate leaf).  This seems a lot of work.

If the developers would consider Markdown as an option for an input format, with the simple display as you'd see in a text editor, here's perhaps another approach to the rendering issue.

I have recently been playing around with org mode in emacs.  There is an export menu to various formats, including plain text, HTML, ODT, LaTeX and PDF.  (All of these are doable [through pandoc](https://github.com/giuspen/cherrytree/issues/878).)  Now the nice thing here is that for all those formats you can also "export and open as" (HTML | ODT | PDF).  That's not quite WYSIWYG, but it's close.  Note that [some PDF readers](https://unix.stackexchange.com/questions/425907/how-to-make-mupdf-automatically-refresh-a-document) will automatically update if the file is updated (or can be made to).  So if we could write in Markdown and export as PDF, that might offer a solution.

So I think I am in agreement with @rwperrott , that while there are a lot of things that keep me with CherryTree, the default richtext format is not one of them.  If we could have a choice to default to some version of Markdown, one with a capacity for including HTML and LaTeX (say pandoc's markdown), then with the ability to export, CherryTree could be much more versatile.
